### DISCUSSION TITLE
Passphrase protected key file for HTTPS Server

### DISCUSSION THREAD

**Discussion by fapifta** [prometheus/prometheus#12447] [Q&A]
Hi,

I would like to set up a prometheus instance, that provides its UI/API endpoint over https.
As I see, it is possible to set up a cert_file and a key_file in the `web.config.file`, however when I have a private key file that is protected by a passphrase, I can not supply the passphrase in any ways. (I hope it is just me and there is a way.)

My problem is that internal policies require me to have a passphrase protected key file, and supply the passphrase in a way that it is accessible only in memory, so that I can not just put that as plain text into the config files either even if it works, and also because of the policy I can not use a plain key file in PEM format without password protection.
The best I can do is to provide a small ramdisk and via that I can provide the password as a file that contains the password, or I can specify it in an environment variable from which Prometheus can get it.

Is there a way to achieve what I would like to do? If so how?

note: I guess this is similar/related to what is described in [Issue #3678](https://github.com/prometheus/prometheus/issues/3678).

**Comment by dombizita** [comment]
Hi, could you maybe look into this @roidelapluie? I saw that you worked on #8316. Thanks in advance!
